### PR TITLE
Polish SafeAccess extension methods to use getters and support Iterable.

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Comment by jacob314** [comment]
Fyi filed 
https://github.com/dart-lang/sdk/issues/40713 
to track the crasher bug in the CFE that the bug in utils_test.dart bug triggered.

File: packages/devtools_app/lib/src/utils.dart [hunk 1]
```diff
@@ -550,3 +542,13 @@ extension SafeAccess<T> on List<T> {
     }
   }
 }
+
+extension SafeAccess<T> on Iterable<T> {
+  T get safeFirst {
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/utils.dart]: nit:
T get safeFirst => isNotEmpty ? first : null;
T get safeLast => isNotEmpty ? last : null;

File: packages/devtools_app/test/utils_test.dart [hunk 2]
```diff
@@ -417,25 +417,29 @@ void main() {
 
     group('SafeAccess', () {
       test('safeFirst', () {
-        final list = [];
-        expect(list.safeFirst(), isNull);
+        final list = <int>[];
+        final Iterable<int> iterable = list;
```
Comment (terrylucas) [inline packages/devtools_app/test/utils_test.dart]: nit: Do we need the Iterable<int> on lvalue with list = <int>[] inference?

Comment (jacob314) [reply packages/devtools_app/test/utils_test.dart]: the cast to I do want to test on Iterable to make sure the extension methods are applying to Iterable as well as List. Maybe that doesn't need to be tested but i think it is a reasonable sort of thing to verify given that extension method behavior does vary depending on the type of the target.

File: packages/devtools_app/lib/src/utils.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/utils.dart]: done. also made safeGet a one liner.

**Pull request merged**